test/testdata/compiler/exceptions/ensure_crash__1.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
```

Comment:
This test is a reproducer (we think) for the segfaults we got from #4473, right? Could we add a comment to that effect here, and maybe move it to `testdata/compiler/regression` (or, idk, maybe `testdata/compiler/exceptions/regression`? :))